app/system/survey/js/databaseUtils.js

```diff
@@ -563,6 +563,12 @@ return {
                 // convert from a nanosecond-extended iso8601-style LOCAL TIME ZONE time HH:MM:SS.sssssssss
                 // this does not preserve the nanosecond field...
                 return odkCommon.toDateFromOdkTime(new Date(), value);
+            } else if ( jsonType.elementType === 'date_year_only' ) {
+                return moment(value).year();

```

Comment:
I think the previous code you had here might be better. It looks like `year()` returns a number whereas `format('YYYY')` was a string

```
> moment('2000-07-01').format('YYYY')
"2000"

> moment('2000-07-01').year()
2000
```

This will probably work most (maybe all?) of the time anyway because of Javascript type coercion though, so not that important to change it